pkgs/tools/misc/byobu/default.nix

```diff
@@ -11,26 +16,52 @@ stdenv.mkDerivation rec {
 
   doCheck = true;
 
-  buildInputs = [ python3 perl ];
-  propagatedBuildInputs = [ textual-window-manager ];
+  buildInputs = [ perl pythonEnv makeWrapper ];
+  propagatedBuildInputs = [ textual-window-manager screen pythonEnv ];
 
-  meta = {
+  postUnpack = ''
+    substituteInPlace $sourceRoot/usr/bin/byobu-select-profile.in \
+      --replace "/bin/true" "true" \
+      --replace "gettext" "${gettext}/bin/gettext"
+    substituteInPlace $sourceRoot/usr/bin/byobu-export.in \
+      --replace "gettext" "${gettext}/bin/gettext"
+  '';
+
+  postInstall = ''
+    # Override the symlinks otherwise they mess with the wrapping
+    cp --remove-destination $out/bin/byobu $out/bin/byobu-screen
+    cp --remove-destination $out/bin/byobu $out/bin/byobu-tmux
+
+    for i in $out/bin/byobu*; do
+      # We don't use the usual ".$package-wrapped" because arg0 within the shebang scripts
+      # points to the filename and byobu matches against this to know which backend
+      # to start with
+      file=".$(basename $i)"
+      mv $i $out/bin/$file
+      makeWrapper "$out/bin/$file" "$out/bin/$(basename $i)" --argv0 $(basename $i) \
+        --set BYOBU_PATH "${pythonEnv}/bin/python" \
+        --set BYOBU_PYTHON "${pythonEnv}/bin/python" \
+        --prefix PATH : ${lib.escapeShellArg (lib.makeBinPath [ vim bc pythonEnv ])}
```

Comment:
I'm not entirely sure what `byobu` is exactly, but I think its some kind of terminal-window manager right? If so, you would use it interactively and launch other software through it? In that case prefixing the `pythonEnv` to `PATH` might be very confusing since it overrides the system python.

Reply:
`PATH` was merged with `BYOBU_PATH` since it made no sense to have both